lib/index.js

```diff
@@ -31,7 +26,7 @@ export const reduceNode = (state, action) => {
   switch (type) {
     case SET_PEERS: {
       assert(Array.isArray(payload), 'Payload for SET_PEERS must be array');
-      return state.set('peers', payload);
+      return Object.assign({}, state, { peers: payload });
```

Comment:
Any preference towards using spread vs. `Object.assign`? Really just a style thing but might be good to keep it consistent. I like the pattern of setting a `newState` at the beginning of the switch and then using that throughout. 
e.g. `const newState = { ...state }; `

Reply:
Like this?

```
const newState = { ...state };
Object.assign(newState, { peers: payload });
```

Reply:
You could also just do `newState.peers = payload`. The idea being is you've made a copy so not changing the original state and so can treat it as mutable.

Reply:
Ah ok i see, do you have a strong preference for that style? I lean towards the more functional way just because it feels less likely to introduce a bug but if you have a strong preference for the copy + mutation, I'll go with that - I did a copy + mutation with removing a key from an object because it was less lines of code than writing a loop + conditional assignment

Reply:
Yeah, I can see the advantage of avoiding the mutation. With a simple switch this way definitely is less code, but if you have many all you have to do is update `newState` in each case and return newState at the end which is why I think this is a common pattern.